### ISSUE TITLE
[EN-Google] No actualization, even not after new install, stuck in "Data is being downloaded..." / DE: Aktualisiert nicht mehr und nach Neuinstalation Daten laden endet nicht

### ISSUE THREAD

**Issue by devaxel** [corona-warn-app/cwa-app-android#2081]
EDIT - Corona-Warn-App Open Source Team

- see blog [Issue Fixed](https://www.coronawarn.app/en/blog/2021-01-14-android-key-matching-issue-fixed/)
- see blog [Issue Reported](https://www.coronawarn.app/en/blog/2021-01-13-corona-warn-app-key-matching-issue-on-android-smartphones/)
- see comment [Issue reported to Google](https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759343313)

- see comment [Issue Fixed by Google](https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759964666)

> The Google Fix should be rolled out to all devices now, all device should recover automatically.
> 
> Google Statement:
> We are aware of an issue with the provideDiagnosisKeys() method of the Exposure Notifications API which affects all apps using the API. The issue became widespread around 12 Jan 22:00 UTC, and it causes the provideDiagnosisKeys() to appear to hang.
> Tl;dr;
> There is an issue currently affecting all users that prevent apps from matching the latest exposure data (since Jan 12th, 2021), but not from continuing broadcasting or scanning keys to devices in proximity
> A fix is rolling out and will reach over 90% of your users by 1/14/2021 AM, EST time.
> Users should not manually clear cache or storage, or uninstall the app. This won’t fix the issue, and it will incur exposure data loss on their devices
> 
> What is happening?
> 
> Calls to provideDiagnosisKeys() never return due to a defect originated on Jan 12th 2021, and in consequence matching does not occur. As a result, Exposure Notifications apps can’t match the latest exposure data (since Jan 12th only). This is the only EN API function affected, and we can confirm that:
> 
> Apps are handling exposure data in the device uninterrupted (broadcasting and scanning)
> Uploading keys is working correctly and was never interrupted (reporting a case through the app)
> 
> We have determined the root cause, and a fix is already rolling out to users’ devices. Once the fix is installed in the device, calls to provideDiagnosisKeys() will have the expected behavior again. The rollout of the fix will reach over 90% of your users within 24 hrs.

---
Duplicates: https://github.com/corona-warn-app/cwa-app-android/issues/2091, https://github.com/corona-warn-app/cwa-app-android/issues/2092
Possibly related to: https://github.com/corona-warn-app/cwa-app-android/issues/2093

ORIGINAL POST
---

Da die App den Risikostatus seit 36 Stunden nicht aktualisierte habe ich sie deinstalliert. Anschließend neu installiert und gestartet. Nach Erteilung aller Berechtigungen zeigt sie nun an "Lade Daten. Das kann ein paar Minuten dauern." Das zeigt sie nun seit 3 Stunden an.
Handy: Galaxy S10
Android 10, alle Updates
App Version: 1.10.1

---
Internal Tracking ID: [EXPOSUREAPP-4594](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4594) duplicate of
Internal Tracking ID: [EXPOSUREAPP-4592](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4592)

**Comment by MikeMcC399** [comment]
I am having the same issue on two different Android devices
CWA shows "Data is being downloaded..." for a long time. I restarted the devices and they are not trying to do an exposure check again. 

One device is connected via WiFi, the other via mobile data.

Other network services are working correctly.

Settings > Google > COVID-19 Exposure Notifications shows a screen without the count of exposure checks. 
The exposure checks are however still accessible via the three-dot overflow menu which offers the two options: Exposure checks and Delete random IDs. There is currently no option displayed to export exposure checks.
![image](https://user-images.githubusercontent.com/66998419/104422245-c8e9c000-557c-11eb-8aa4-287c1d8be3de.png)

**Comment by dsarkar** [comment]
@devaxel, @MikeMcC399 

Thanks for report. Is this a new issue, suddenly, apparently not related to any CWA update?

**Comment by Ein-Tim** [comment]
Not reproducible one my iOS devices.

**Comment by fhausmann** [comment]
Same issue since yesterday (last successful check yesterday afternoon) on
Pixel 2 XL, Android 11, App Version 1.10.1
and Honor 10 Lite, Android 10, App Version 1.10.1

No CWA update made in > 3 days.

**Comment by dsarkar** [comment]
@fhausmann Thanks for report.

FYI - we have similar reports today on the play store.

**Comment by MikeMcC399** [comment]
@dsarkar 
The issue is happening today. The last successful exposure check was yesterday 12 JANUARY 23:18 on one device and yesterday 12 JANUARY 18:28 on another device.

https://allestörungen.de/ shows some signs of network outages for the networks I am using:

- Deutsche Telekom (WLAN) https://xn--allestrungen-9ib.de/stoerung/deutsche-telekom/
- Vodafone (mobile network) https://xn--allestrungen-9ib.de/stoerung/vodafone/
- PŸUR (WLAN) no outage shown

When I ping svc90.main.px.t-online.de I get
```
Pinging svc90.main.px.t-online.de [2003:2:20:704:87:140:208:122] with 32 bytes of data:
Reply from 2003:2:20:704:87:140:208:122: time=17ms
```
so there is basic connectivity available.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
When was your last exposure check?
How are you connecting to Internet?

**Comment by dsarkar** [comment]
Internal ticket ID: EXPOSUREAPP-4594

**Comment by dsarkar** [comment]
Actually in my iPhone 6s, CWA 1.10.1, iOS 14.3, last update 02:13 today.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

Last check was today at 05:45, will try again in 4 Minutes and report back. Connected via WiFi.

**Comment by Ein-Tim** [comment]
Everything works for me (iOS 14.4 Public Beta 1, iPhone 6s)

**Comment by dsarkar** [comment]
FYI - I have been told, that devs are in contact with google. Might be ENF related on some devices. Being investigated now.

**Comment by nc-kano** [comment]
The same issue happens in Danish and Norwegian apps. It looks like a global issue.

**Comment by Ein-Tim** [comment]
@dsarkar 
So do you experience this Issue?

**Comment by dsarkar** [comment]
@Ein-Tim right now on mobile data, will check later on wifi.

**Comment by Jo-Achim** [comment]
Hello, 

same issue here with the same 'Covid-19-Benachrichtigung' like https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759273658.

Last Update 12.01.2012 15:25 (yesterday).


Android 10 with Android security update: December 1st, 2020
Nokia 7.1, Build: 00WW_4_16B_SP02
CWA version: 1.10.1
ENF: 18204714000
NetGuard Pro 2.290
Norton 360, v.5.0.0.5437

**Comment by Ein-Tim** [comment]
Okay so with WLAN it worked for me, I will check with mobile data later.

Still you could transfer this to the documentation repo since it seems like iOS devices are also affected...

**Comment by dsarkar** [comment]
@MikeMcC399 let me know your device and android version, please. Creating a list of devices and  android versions.

**Comment by MikeMcC399** [comment]
@dsarkar 
All devices are running Exposure Notification System: 18204714000
Android 8 and 10

> FYI - I have been told, that devs are in contact with google. Might be ENF related on some devices. Being investigated now.

also from @nc-kano 
> The same issue happens in Danish and Norwegian apps. It looks like a global issue.

I also tried installing and switching to the Irish app. I got no exposure checks logged in Google with the Irish app active, so I can imagine that there is a configuration issue with Google. We know that Google passes some parameters to the Exposure Notification System.

**Comment by dsarkar** [comment]
@MikeMcC399 Thanks for details.

@devaxel, @fhausmann Your ENF is 18204714000, as is in the case of @MikeMcC399 and @Jo-Achim ? Thanks.

**Comment by fhausmann** [comment]
Yes

**Comment by solaris7590** [comment]
Same issue here. Last update yesterday 18:36.

Pixel 3XL
Android 11, Security Update Dec. 2020
CWA Version: 1.10.1
Google "COVID-19-Benachrichtigung" Version: 18204714000.

I'm not sure, but for me it seems that Google updated the "COVID-19-Benachrichtigung". Maybe there is some connection...

**Comment by nc-kano** [comment]
In Danish app we have confirmation from Google:
"Thank-you! Yes we're unfortunately seeing a few instances of seemingly similar behaviour across ENS apps. Can you provide an adb bugreport? Perhaps take one now, and then also after a reboot (it seems likely to persist after a reboot and the bugreport will be "cleaner" after a reboot)"
Are anyone of you who experienced the issue able to generate such report?

**Comment by uschindler** [comment]
Hi, same issue here on 2 different Samsung Galaxy S7 phones.

It either hangs at downloading or at the exposure checks.

My last check was yesterday at 16:38, for my wife's phone a bit later at around 18:00. Since this time no updates anymore and after a restart of app it hangs endless. If you restart phone it does nothing, until the next check is to be done.

It looks like Google Play Services were updated for many phones during the last days, so this issue does not affect all users at same time.

As noted before the UI of the Google Settings about Exposure Checks changed. The list of checks is now hidden behind the burger menu top right.

My GPS version:
![20210113_102142](https://user-images.githubusercontent.com/1005388/104433097-fab55380-5589-11eb-9ab1-31148198ab3c.jpg)

**Comment by MikeMcC399** [comment]
@nc-kano 
> Are anyone of you who experienced the issue able to generate such report?

These are the instructions to generate a bug-report
https://developer.android.com/studio/debug/bug-report

Be aware that Personally Identifiable Information may be collected by a bug-report, so it is not recommended to share it in a public forum. Only make it available to somebody / some organisation you trust with your data.

**Comment by thomasaugsten** [comment]
At the moment there is no exposure check possible with the Google ENF on android devices. We informed Google about this issue and they are working on a fix. We receive this issue also from multiple other countries. 
When the problem is fixed from google side the app should recover themself with the next background check.
We will inform you when we receive updates.

**Comment by Ein-Tim** [comment]
@dsarkar You could add the bug lable

**Comment by Ein-Tim** [comment]
@thomasaugsten iOS is working correctly, or?

**Comment by thomasaugsten** [comment]
We see no issues on iOS at the moment.

**Comment by dsarkar** [comment]
@Ein-Tim Indeed, I confirm, now with Wifi updated are done on my iOS device
.

**Comment by MikeMcC399** [comment]
PR https://github.com/corona-warn-app/cwa-website/pull/768 plans to publish to blog.

**Comment by Ein-Tim** [comment]
Here it is: https://www.coronawarn.app/en/blog/2021-01-13-corona-warn-app-key-matching-issue-on-android-smartphones/

**Comment by vaggelisdouros** [comment]
same issue for me:

last check: yesterday at 17:12

Specs:
samsung galaxy a5 (2016) running android 7
ENF: 18204714000

**Comment by FrankfurterRadler** [comment]
Same issue here:
Last check yesterday at 22:29
Samsung Galaxy A5 (2016)
Android 7.0
CWA 1.10.1
ENF 18204714000

**Comment by vaggelisdouros** [comment]
BREAKING NEWS:

In my device (see 2 comments above), the update just worked.

**Comment by MikeMcC399** [comment]
@vaggelisdouros 
> In my device (see 2 comments above), the update just worked.

I am seeing on two different devices respectively:

1. "Check is running.." which then resulted in "Updated: Yesterday, 23:19" (instead of current time) and the Google exposure check log still has the appearance as in https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759273658
2. "Data is being downloaded..." without this status being completed.

So, for me, there is not yet a resolution.

**Comment by MikeMcC399** [comment]
@dsarkar 
Would it make sense to pin this hot topic so it show up even more visible at the top of the issue list?

**Comment by Ein-Tim** [comment]
@thomasaugsten Is there already any assumption how an Issue like this could happen?

**Comment by tangomanni** [comment]
CWA did no update this morning , but few minutes ago at 16:33 an update of risk calculation
![Screenshot_20210113-164121_Corona-Warn](https://user-images.githubusercontent.com/73969345/104475692-b0e76000-55bf-11eb-94a8-a7f2f41e3885.jpg)
 was done

**Comment by thomasaugsten** [comment]
Could you please reboot your device and start the Corona warn app

**Comment by codejus** [comment]
@thomasaugsten  Rebooted the phone, opened CWA and had opened it for 5 min. Nothing has changed, still last checked yesterday at 20:16. Not entry in the Exposure Log either.

**Comment by MikeMcC399** [comment]
@thomasaugsten 
After restarting on Android 8, CWA continued to show
"Encounters have not yet been checked."

Since this is just a test device I will uninstall and re-install CWA to see if it recovers.

**Comment by thomasaugsten** [comment]
No reinstall is needed only be patient to receive the fix from google on your device

**Comment by Ein-Tim** [comment]
@thomasaugsten 
Is Google rolling out a fix at the moment?

**Comment by MikeMcC399** [comment]
@thomasaugsten 
Reinstall (not recommended for other users!) got the exposure check working.

Is the change to the Google UI now permanent?

Previously 
Settings > Google > COVID-19 Exposure Notifications
would show the number of exposure checks and the timestamp of the last exposure check.
This information is now missing at this level of the UI.
Instead it is possible to tap the three-dot overflow menu which contains the menu items:
- Exposure checks
- Delete random IDs

From the exposure check display, once again tapping the three-dot overflow menu gives access to:
- Export exposure checks
- Delete exposure checks

Was this change of UI part of the underlying cause which temporarily broke exposure checking?

**Comment by Jo-Achim** [comment]
Yes, this way works, _but not completely_.

Settings > Google > COVID-19 Exposure Notifications
I called "Exposure checks" (3 dots-menue) twice and 'open' Corona-Warn-App. 
Now: Update: Today, 18:52.

**But:** without the count of exposure checks; see https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759273658.


Android 10 with Android security update: December 1st, 2020
Nokia 7.1, Build: 00WW_4_16B_SP02
CWA version: 1.10.1
ENF: 18204714000
NetGuard Pro 2.290
Norton 360, v.5.0.0.5437

**Comment by MikeMcC399** [comment]
After just waiting a couple of hours the exposure checking has now recovered on its own and the main screen of the app is showing Updated: **Today**, hh:mm. Rebooting did not speed up the recovery for me, but others may have different outcomes.

**Comment by ugroempi** [comment]
For me, the issue did not go away on its own. But after forcing a stop of the app and then re-opening it, the check ran normally again (for now) and also yielded a current status. I did not reboot the phone (Android 10, Samsung phone).

**Comment by heinosahling** [comment]
![Screenshot_20210113_224932_com google android gms](https://user-images.githubusercontent.com/66993577/104514702-f6bd1c00-55f1-11eb-938d-4e747817c699.jpg)

These are the checks on my phone. Today all manual.
Something went wrong yesterday in the evening.

**Comment by thomasaugsten** [comment]
The Google Fix should be rolled out to all devices now.

**Google Statement:**
We are aware of an issue with the provideDiagnosisKeys() method of the Exposure Notifications API which affects all apps using the API. The issue became widespread around 12 Jan 22:00 UTC, and it causes the provideDiagnosisKeys() to appear to hang. 
Tl;dr;
There is an issue currently affecting all users that prevent apps from matching the latest exposure data (since Jan 12th, 2021), but not from continuing broadcasting or scanning keys to devices in proximity
A fix is rolling out and will reach over 90% of your users by 1/14/2021 AM, EST time.
Users should not manually clear cache or storage, or uninstall the app. This won’t fix the issue, and it will incur exposure data loss on their devices

What is happening?

Calls to provideDiagnosisKeys() never return due to a defect originated on Jan 12th 2021, and in consequence matching does not occur. As a result, Exposure Notifications apps can’t match the latest exposure data (since Jan 12th only). This is the only EN API function affected, and we can confirm that:

Apps are handling exposure data in the device uninterrupted (broadcasting and scanning)
Uploading keys is working correctly and was never interrupted (reporting a case through the app)

We have determined the root cause, and a fix is already rolling out to users’ devices. Once the fix is installed in the device, calls to provideDiagnosisKeys() will have the expected behavior again. The rollout of the fix will reach over 90% of your users within 24 hrs.

**Comment by MikeMcC399** [comment]
@thomasaugsten 
Thank you very much for providing the statement and information from Google!

> Calls to provideDiagnosisKeys() never return due to a defect originated on Jan 12th 2021 ...

There was no change to the version number shown in 
Settings > Google > COVID-19 Exposure Notifications
on Jan 12, 2021 or yesterday Jan 13, 2021 when many users first noticed the problem and reported here in this thread that they were using Exposure Notifications System (ENS) version 18204714000.

How can Google change the functionality of Exposure Notifications System without changing its version number?

> We have determined the root cause, and a fix is already rolling out to users’ devices. Once the fix is installed in the device, calls to provideDiagnosisKeys() will have the expected behavior again. The rollout of the fix will reach over 90% of your users within 24 hrs.

How is it possible to determine by looking at the mobile device software component version numbers whether the fix has been installed? If this is not possible then I have concerns about the supportability of the CWA solution which depends heavily on the correct functioning of Google ENS on Android devices.


There is no update to the release notes on https://developers.google.com/android/exposure-notifications/release-notes concerning the current version 1.8 of ENS. Do you expect Google to publish release information about the current version on that site, which shows "Last updated 2020-10-02 UTC."

**Comment by CDernbach** [comment]
> EDIT - Corona-Warn-App Open Source Team
> 
> * see blog https://www.coronawarn.app/en/blog/2021-01-13-corona-warn-app-key-matching-issue-on-android-smartphones/
> * see comment [#2081 (comment)](https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759343313)
> 
> > At the moment there is no exposure check possible with the Google ENF on android devices. We informed Google with bug reports about this issue and they are working on a fix. We receive this issue also from multiple other countries. When the problem is fixed from google side the app should recover themself with the next background check.
> > We will inform you when we receive updates.
> 
> Duplicates: #2091, #2092
> Possibly related to: #2093
> 
> ## ORIGINAL POST
> Da die App den Risikostatus seit 36 Stunden nicht aktualisierte habe ich sie deinstalliert. Anschließend neu installiert und gestartet. Nach Erteilung aller Berechtigungen zeigt sie nun an "Lade Daten. Das kann ein paar Minuten dauern." Das zeigt sie nun seit 3 Stunden an.
> Handy: Galaxy S10
> Android 10, alle Updates
> App Version: 1.10.1
> 
> Internal Tracking ID: [EXPOSUREAPP-4594](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4594) duplicate of
> Internal Tracking ID: [EXPOSUREAPP-4592](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4592)

Statement from Google to German News-Agency dpa (Deutsche Presse-Agentur):

"Late in the evening on January 12, an issue with the Exposure Notifications System on Android began causing delays in the checking of potential exposures for those with apps  installed. We have issued a fix. It may take a few hours for devices to catch up, and in some cases we will work directly with developers to help with recovery. The issue did not cause the loss of any data or potential exposures.”

**Comment by MikeMcC399** [comment]
There is a follow-up blog post https://www.coronawarn.app/en/blog/2021-01-14-android-key-matching-issue-fixed/ on this issue.

**Comment by Jo-Achim** [comment]
> There was no change to the version number shown in
> Settings > Google > COVID-19 Exposure Notifications
> on Jan 12, 2021 or yesterday Jan 13, 2021 when many users first noticed the problem and reported here in this thread that they were using Exposure Notifications System (ENS) version 18204714000.
> 
> **How can Google change the functionality of Exposure Notifications System without changing its version number?**

Very right, if not even a version numbering is handled properly ...!?

**Comment by MikeMcC399** [comment]
@Jo-Achim 
> Very right, if not even a version numbering is handled properly ...!?

I fired up an Android 11 emulator which I hadn't used for a week or two and it was showing 1820**45**16000 with the new menu for Google Exposure Checks (the one which no longer explicitly shows the last time updated as in https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759273658). I manually updated Google Play services to the latest version 20.47.14 (150700-349456378) which updated the Exposure Notification System (ENS) version displayed to 18204714000. The menu stayed the same.

If the new menu showed up even on 20.45.16 of Google Play services then it seems even more likely that some components of Google ENS are updated via a silent install independently of the version of Google Play services. It would be good to understand this mechanism better. The FAQ https://www.coronawarn.app/en/faq/#ENF_version doesn't cover this aspect in enough detail to explain the observed behaviour.

**Comment by DooMMasteR** [comment]
I have been updated to 1820**50**14000 now... and everything seems to be back to normal for me.

**Comment by Jo-Achim** [comment]
Has anyone received the fix that Google promised?
I haven't received anything up to now (01/15/2021 18:05), i.e. I still have ENF: 18204714000 with the above-mentioned problems.

Or unfortunate translation? Blog entry from January **14**, 2021 ([https://www.coronawarn.app/de/blog/2021-01-14-android-key-matching-issue-fixed/](url)): "The Corona warning app should work again on the corresponding devices **this morning**."

**Comment by MikeMcC399** [comment]
@Jo-Achim 
It seems that the fix does not depend on or change the version number. So it would be good to know the answer to: "How to determine if the fix is installed?"

**Comment by codejus** [comment]
@Jo-Achim I have the same ENF Version as you on two of my phones, but CWA is working normal (again) on my Nokia 7 Plus (Since the evening of the 13th) and freshly installed Pixel 4a 5G. Both are runnung with PlayService 20.47.14.
So the fix is not connected to a version upgrade. Probably they've just pulled the faulty version of 18204714000 back and rereleased it with the fixes or it was a configration issue at all, which got fixed silently.

@MikeMcC399 The only determination I see is that its's working again or not, as the root cause is unknown.

**Comment by MikeMcC399** [comment]
@codejus 
> The only determination I see is that its's working again or not, as the root cause is unknown.

Yes, that is the only way we know at the moment.

https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759964666 quoted from Google says:
"We have determined the root cause ..."

It would be good for our understanding if the knowledge about the root cause were to be shared with the community.

**Comment by Jo-Achim** [comment]
So, besides the question of how we can tell if the fix is currently installed ...
In addition, the question arises whether the problem of: "Exposure Notifications shows a screen without the count of exposure checks." shouldn't have also been fixed? Because my screenshot still looks like this:
![Screenshot_20210115-190955](https://user-images.githubusercontent.com/21124877/104764141-4113d980-5767-11eb-9ac0-ccc6ea8c31bf.png)

The fact that CWA fundamentally updates the keys again - see: https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759582009 / https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-759620836 - is clear and hopefully generally applicable.

**Comment by sandoxy** [comment]
Last update at: settings->google->covid->check:
today 18:02

Last updaze shown in CWA:
today 10:12

Phone restart dud not change anything.

**Comment by daimpi** [comment]
@Jo-Achim 
> In addition, the question arises whether the problem of: "Exposure Notifications shows a screen without the count of exposure checks." shouldn't have also been fixed?

afaik this is how the new interface of ENF is looking like when working as intended. You can find the exposure checks behind the 3 dot menu in the upper right corner of this screen.

**Comment by Jo-Achim** [comment]
@daimpi, thanks, but that means that the "counter of exposure checks" is no longer available!?

**Comment by daimpi** [comment]
> thanks, but that means that the "counter of exposure checks" is no longer available!?

That's my current understanding, yes… I dunno why and when exactly Google changed this though maybe @MikeMcC399 knows more? 🙂

**Comment by MikeMcC399** [comment]
@daimpi 
We are in the dark about the change to the Google UI and have asked for information. I don't think it is an accident or a bug that it looks different. There have been cases where users have interpreted the count of exposure checks to mean the number of infectious exposures they have been subjected to. I could imagine that Google might have removed the display of the total number of exposure checks because of the danger of misinterpretation. The number is only useful for debugging, not for regular users.

My theory is that the look of the Google UI is changed by a parameter supplied by Google.

One user posted a screen shot in https://github.com/corona-warn-app/cwa-app-android/issues/1817#issuecomment-740516795 showing exactly the "new" screen on Dec 8, 2020 suggesting that the alternate View layout was already loaded onto devices but not generally activated at that date. I also once saw this screen a couple of weeks ago when one of my devices was in a strange state.

**Comment by Jo-Achim** [comment]
Subject: "count of exposure checks".
If the 'Exposure Counter' (see: https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-761117543) is still missing in the future versions of ENF, I suggest the following change to the display in the risk status:
![Screenshot_20210113-094500_3](https://user-images.githubusercontent.com/21124877/104919674-e95bb500-5996-11eb-8204-32657e162016.jpg)


Reason: I consider the exposure counter to be a kind of 'living indicator' that shows that CWA is 'also working in the background'.

**Comment by DooMMasteR** [comment]
I hoped they would add and clarify information, maybe even show the count of beacons in database (because for all I know my phone might not be collecting anything at all, but nope, instead they removed the check counter too.

**Comment by Ein-Tim** [comment]
@DooMMasteR See: https://github.com/corona-warn-app/cwa-wishlist/issues/5

**Comment by DooMMasteR** [comment]
@Ein-Tim das meinte ich nicht wirklich... Es wäre interessant zu wissen, ob das ganze Interface überhaupt funktioniert.
Ich gehe bei mir davon aus, weil der Check selbst Zeit braucht, aber wirklich nachvollziehen kann man es einfach nicht.
Jetzt wo der Check Count sogar wieder verschwunden ist, denke ich mal, dass die Chancen einfach gering sind.

**Comment by Ein-Tim** [comment]
@DooMMasteR 
Okay, maybe you find something what fits your idea here: https://github.com/corona-warn-app/cwa-wishlist/issues/336. 
If not, feel free to [open a new Wishlist Issue](https://github.com/corona-warn-app/cwa-wishlist/issues/new?assignees=&labels=feature+request&template=01_feature_request.md).

**Comment by DooMMasteR** [comment]
It is no issue if the app, as with this issue it is in Google's hands to offer such functionality.
And since communication is... Non optimal... And features seem more to disappear than appear, I would guess our chances to get something like a useful health indicator of the ENF API are actually quite low.

**Comment by Jo-Achim** [comment]
Hello, 
also in the new ENF version 18205016000 the 'Exposure Counter' is missing (see: https://github.com/corona-warn-app/cwa-app-android/issues/2081#issuecomment-762241643)!

Android 10 with Android security update: January 1st, 2021
Nokia 7.1, Build: 00WW_4_16B_SP03
CWA version: 1.10.1
ENF: 18205016000
NetGuard Pro 2.290
Norton 360, v.5.1.0.5628

**Comment by heinezen** [comment]
This was fixed by Google a whiile ago and should not be an issue anymore.

----
Corona-Warn-App Open Source Team
